changelogs/fragments/4026-fix-mail-callback.yml

```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - mail callback plugin - Fix ``'in <string>' requires string as left operand, not int`` Exception.
+  - mail callback plugin - Add ``Message-ID``, ``Date``, ``Content-Type`` headers.
```

Comment:
This is a feature, not a bugfix, and should not be added in the same PR.

Reply:
I’ll move the addition of the `Date` and `Message-ID` headers in a dedicated PR.

Reply:
I’ve opened #4056 to not forget to add the `Date` and `Message-ID` headers, after this PR.
But #4056 will conflict with this PR and will need to be rewritten once this one is merged.
Still, I already opened it to not forget.